kafka-connect-s3/src/main/java/io/confluent/connect/s3/storage/S3OutputStream.java

```diff
@@ -175,6 +182,64 @@ private MultipartUpload newMultipartUpload() throws IOException {
     }
   }
 
+  protected static class RetryWithExponentialBackoff {
+    private int failCount;
+    private int maxRetryAttempts;
+    private Throwable cause;
+    private String errorMsg;
+
+
+    private RetryWithExponentialBackoff(int maxRetryAttempts, String errorMsg) {
+      this.maxRetryAttempts = maxRetryAttempts;
+      this.errorMsg = errorMsg;
+    }
+
+     /**
+      *
+      * @param runnable The method to run with retries
+      * @param retries How many times to retry
+      * @param errorMsg Error message to show
+      * @return Returns either this instance or throws ConnectException if it failed more then given count
+      */
+    public static RetryWithExponentialBackoff blocking(Runnable runnable, int retries, String errorMsg) {
+      RetryWithExponentialBackoff retry = new RetryWithExponentialBackoff(retries, errorMsg);
+      retry.execute(runnable);
+      return retry;
+    }
+
+    private RetryWithExponentialBackoff execute(Runnable runnable) {
+      do {
+        if (failCount > 0) {
+          try {
+            Thread.sleep(100 * 2 ^ failCount);
```

Comment:
Good catch! Was doing some math in R lang and continued the same way in Java :) I replaced it with bit shifting.